tests/io_components/test_dataset.py

```diff
@@ -117,3 +109,22 @@ def test_dataset_get_indices_as_dataframe_predicates():
         columns=["l_external_code"], index=pd.Index([], name="partition")
     )
     pdt.assert_frame_equal(result, expected)
+
+
+def test_dataset_get_indices_as_dataframe_no_index(dataset):
+    assert not dataset.primary_indices_loaded
+    df = dataset.get_indices_as_dataframe()
+    assert not dataset.primary_indices_loaded
```

Comment:
I don't know if we should have this assert in here (and the test below). Imo, ideally we would load the indices in-place but that's another discussion
```suggestion
```